src/tools/perf/lib/ucp_tests.h

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright (c) NVIDIA CORPORATION & AFFILIATES, 2025. ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#pragma once
+
+#ifdef HAVE_CONFIG_H
+#  include "config.h"
+#endif
+
+#include "libperf_int.h"
+
+class ucp_perf_test_runner_base {
+public:
+    ucp_perf_test_runner_base(ucx_perf_context_t &perf) :
+        m_perf(perf)
+    {}
+
+    void request_wait(ucs_status_ptr_t request, ucs_memory_type_t mem_type,
+                      const char *operation_name)
+    {
+        ucs_status_t status;
+
+        if (UCS_PTR_IS_PTR(request)) {
+            do {
+                ucp_worker_progress(m_perf.ucp.worker);
+                status = ucp_request_check_status(request);
+            } while (status == UCS_INPROGRESS);
+            ucp_request_free(request);
+        } else {
+            status = UCS_PTR_STATUS(request);
+        }
+
+        if (status != UCS_OK) {
+            ucs_warn("failed to %s(memory_type=%s): %s", operation_name,
+                     ucs_memory_type_names[mem_type], ucs_status_string(status));
+        }
+    }
+
+protected:
+    ucx_perf_context_t &m_perf;
+};
+
+template<typename PSN>
+class ucp_perf_test_runner_base_psn : public ucp_perf_test_runner_base {
```

Comment:
why need to separate ucp_perf_test_runner_base and ucp_perf_test_runner_base_psn?